mozilla-lockwise/lockwise-android
# 948: prevent duplicate entries during edit

Fixes #948

## Testing and Review Notes

1. `Datastore` contains a list of all entries
2. `Datastore` can filter that list based on a given hostname
	- `fun getUsernamesForDomain(hostname: String): Observable<List<Optional<String>>>`
3. The edit fragment doesn’t want to directly reference the datastore (that's a job for the presenter)
4. Create a local list in the presenter that gets our list of dupes from `getUsernamesForDomain()`
5. Check this list in the fragment on text change and set error state if dupe exists


## Screenshots or Videos

## To Do

- [x] datastore tests
- [x] edit presenter tests
- [x] remove current item from list of possible dupes
- [x] add “WIP” to the PR title if pushing up but not complete nor ready for review
